WordPress/Classes/ViewRelated/Themes/ThemeBrowserViewController.swift

```diff
@@ -0,0 +1,201 @@
+import Foundation
+
+public class ThemeBrowserViewController : UICollectionViewController, UICollectionViewDelegateFlowLayout, NSFetchedResultsControllerDelegate, UISearchBarDelegate {
+    
+    // MARK: - Properties: must be set by parent
+    
+    /**
+     *  @brief      The blog this VC will work with.
+     *  @details    Must be set by the creator of this VC.
+     */
+    private var blog : Blog!
+    
+    // MARK: - Properties: managed object context & services
+    
+    /**
+     *  @brief      The managed object context this VC will use for it's operations.
+     */
+    private let managedObjectContext : NSManagedObjectContext = {
+        ContextManager.sharedInstance()!.newDerivedContext()
+    }()
+    
+    /**
+     *  @brief      The FRC this VC will use to display filtered content.
+     */
+    private lazy var themesController: NSFetchedResultsController = {
+        let fetchRequest = NSFetchRequest(entityName: Theme.entityName())
+        fetchRequest.fetchBatchSize = 20
+        let predicate = NSPredicate(format: "blog == %@", self.blog)
+        fetchRequest.predicate = predicate
+        let sort = NSSortDescriptor(key: "name", ascending: true)
+        fetchRequest.sortDescriptors = [sort]
+        let frc = NSFetchedResultsController(fetchRequest: fetchRequest, managedObjectContext: self.managedObjectContext,
+            sectionNameKeyPath: nil,
+            cacheName: nil)
+        frc.delegate = self
+        
+        do {
+            try frc.performFetch()
+        } catch let error as NSError {
+            DDLogSwift.logError("Error fetching themes: \(error.localizedDescription)")
+        }
+        
+        return frc
+    }()
+    
+    /**
+     *  @brief      The themes service we'll use in this VC.
+     */
+    private lazy var themeService : ThemeService = {
+        ThemeService(managedObjectContext: self.managedObjectContext)
+    }()
+    
+    // MARK: - Additional initialization
+    
+    public func configureWithBlog(blog: Blog) {
+        do {
+            let blogInContext = try managedObjectContext.existingObjectWithID(blog.objectID) as? Blog
+            self.blog = blogInContext
+        } catch let error as NSError {
+            DDLogSwift.logError("Error finding blog: \(error.localizedDescription)")
+        }
+    }
+    
+    // MARK: - UIViewController
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        title = NSLocalizedString("Themes", comment: "Title of Themes browser page")
+        
+        WPStyleGuide.configureColorsForView(view, collectionView:collectionView)
+        
+        updateThemes()
+    }
+
+    public override func didRotateFromInterfaceOrientation(fromInterfaceOrientation: UIInterfaceOrientation) {
```

Comment:
This method was deprecated in iOS 8. Since we're bumping the deployment target to 9... maybe `viewWillTransitionToSize` ?